--- a/src/interface.c
+++ b/src/interface.c
@@ -1680,12 +1680,16 @@
           if (GPUINFO_DYNAMIC_FIELD_VALID(&device->dynamic_info, gpu_clock_speed) &&
               GPUINFO_DYNAMIC_FIELD_VALID(&device->dynamic_info, gpu_clock_speed_max)) {
             data_val = device->dynamic_info.gpu_clock_speed * 100 / device->dynamic_info.gpu_clock_speed_max;
+            if (data_val > 100)
+              data_val = 100u;
           }
           break;
         case plot_gpu_mem_clock_rate:
           if (GPUINFO_DYNAMIC_FIELD_VALID(&device->dynamic_info, mem_clock_speed) &&
               GPUINFO_DYNAMIC_FIELD_VALID(&device->dynamic_info, mem_clock_speed_max)) {
             data_val = device->dynamic_info.mem_clock_speed * 100 / device->dynamic_info.mem_clock_speed_max;
+            if (data_val > 100)
+              data_val = 100u;
           }
           break;
         case plot_information_count:
