--- a/src/interface.c
+++ b/src/interface.c
@@ -2142,6 +2142,8 @@
               GPUINFO_DYNAMIC_FIELD_VALID(&device->dynamic_info, gpu_clock_speed_max) &&
               device->dynamic_info.gpu_clock_speed_max > 0) {
             data_val = device->dynamic_info.gpu_clock_speed * 100 / device->dynamic_info.gpu_clock_speed_max;
+            if (data_val > 100)
+              data_val = 100u;
           }
           break;
         case plot_gpu_mem_clock_rate:
@@ -2149,6 +2151,8 @@
               GPUINFO_DYNAMIC_FIELD_VALID(&device->dynamic_info, mem_clock_speed_max) &&
               device->dynamic_info.mem_clock_speed_max > 0) {
             data_val = device->dynamic_info.mem_clock_speed * 100 / device->dynamic_info.mem_clock_speed_max;
+            if (data_val > 100)
+              data_val = 100u;
           }
           break;
         case plot_effective_load_rate:
